### DISCUSSION TITLE
Does bleak provide better method to check connection?

### DISCUSSION THREAD

**Discussion by snowuyl** [hbldh/bleak#935] [Q&A]
* bleak version: 0.15.0a1
* Python version: 3.10.6
* Operating System: Windows 10

### Description
I am using is_connected() API to check if client is connecting to Nordic device. Does bleak provide better method to check connection? 
async with BleakClient(ble_device.address, loop=loop) as client:
    # waiting for BLE client to be connected
    x = await client.is_connected()

**Comment by dlech** [comment]
The device will already be connected inside of the `async with` block, so there is no need to check `is_connected`. You can also add a `disconnected_callback` if you want to know when the device disconnects.

**Reply by snowuyl** [reply]
Thanks for your reply! Does bleak provide example to add disconnect_callback? If disconnected from Nordic device, how bleak knows that connecting is disconnected?
